packages/contracts/contracts/MVM/MVM_InboxSenderManager.sol

```diff
@@ -0,0 +1,111 @@
+    string public constant CONFIG_OWNER_KEY = "METIS_MANAGER";
+
+    /*************
+     * Variables *
+     *************/
+    // blockNumber => InboxSenderType => inboxSender
+    mapping(uint256 => mapping(InboxSenderType => InboxSender)) public inboxSenders;
+
+    uint256[] public blockNumbers;
+    mapping(InboxSenderType => address) public defaultInboxSender;
+
+    /***************
+     * Constructor *
+     ***************/
+    constructor(address _libAddressManager, InboxSender[] memory _defaultInboxSenders)
+        Lib_AddressResolver(_libAddressManager)
+    {
+        for (uint256 i = 0; i < _defaultInboxSenders.length; ++i) {
+            defaultInboxSender[_defaultInboxSenders[i].senderType] = _defaultInboxSenders[i].sender;
+        }
+    }
+
+    /**********************
+     * Function Modifiers *
+     **********************/
+    modifier onlyManager() {
+        require(
+            msg.sender == resolve("METIS_MANAGER"),
+            "MVM_InboxSenderManager: Function can only be called by the METIS_MANAGER."
+        );
+        _;
+    }
+
+    /********************
+     * Public Functions *
+     ********************/
+    function setInboxSenders(uint256 blockNumber, InboxSender[] calldata _inboxSenders)
+        external
+        override
+        onlyManager
+    {
+        _setInboxSenders(blockNumber, _inboxSenders);
+    }
+
+    // allow us to overwrite the last block number and its senders, just in case if we made any mistake
+    function overwriteLastInboxSenders(uint256 blockNumber, InboxSender[] calldata _inboxSenders)
+        external
+        override
+        onlyManager
+    {
+        require(blockNumbers.length > 0, "MVM_InboxSenderManager: No block to update.");
+
+        // pop the last block
+        uint256 lastBlockNumber = blockNumbers[blockNumbers.length - 1];
+        blockNumbers.pop();
+
+        // clean up the last block senders
+        mapping(InboxSenderType => InboxSender) storage lastInboxSenders = inboxSenders[lastBlockNumber];
+        delete lastInboxSenders[InboxSenderType.InboxSender];
+        delete lastInboxSenders[InboxSenderType.InboxBlobSender];
+
+        // write the new senders
+        _setInboxSenders(blockNumber, _inboxSenders);
+    }
+
+    function getInboxSender(uint256 blockNumber, InboxSenderType inboxSenderType) external view override returns (address) {
+        for (int256 i = int256(blockNumbers.length) - 1; i >= 0; i--) {
+            if (blockNumbers[uint256(i)] <= blockNumber) {
+                address sender = inboxSenders[blockNumbers[uint256(i)]][inboxSenderType].sender;
+                if (sender != address(0)) {
+                    return sender;
+                }
+            }
+        }
+
+        return defaultInboxSender[inboxSenderType];
+    }
+
+    /********************
+     * Internal Functions *
+     ********************/
+    function _setInboxSenders(uint256 blockNumber, InboxSender[] calldata _inboxSenders)
```

Comment:
You check if the _inboxSenders' length is not zero

Reply:
Fixed